### ISSUE TITLE
Sprint November 13 - November 24

### ISSUE THREAD

**Issue by timgl** [PostHog/posthog#18460]
# Global Sprint Planning

### from previous sprint issues
 
solved/delete/or carry forward?

* Clickhouse EU/cloud
* Terraform - James to schedule an incident analysis
* Clickhouse visibility - had an incident, and then we had to sudo command in.
* Gunicorn -> Frank looking at killing both

1.
2.
3.

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by mariusandra** [comment]
# Project Query Engine

**Support hero:** Tom, followed by Marius

## Retro

Only Tom was working the first week. Second week Julian's onboarding in Cambridge

- Trends insights continued (aggregations) @Gilbert09
  - Feature complete, not released
- Have a great onboarding week
   - In progress
- Async slow lane for queries that take 60 < x < 600 seconds to complete.
   - Didn't get to this.

## Hang over items from previous sprint

- Insight refactor, continued.

## OKR

- Project "Finish the darned Query Engine" (Tom, Julian, Marius)
  - 🟡 -> 🟢 Finish insight HogQL conversion. 
  - 🔴 -> 🟡 Support non-event data sources on HogQL insights (BI tools for data warehouse).
  - 🔴 -> 🔴 Build tools to measure HogQL query performance, and catch regressions before they hit users.

### High priority

- Launch HogQL trends and lifecycle insights @Gilbert09 @mariusandra 
   - Finish trends
   - Check for data integrity between the old and new systems.
- Start building the HogQL BI features @Gilbert09 
   - Trends graphs from any HogQL query
   - Replicating the go/revenue dashboard.
- Async slow lane for queries that take 60 < x < 600 seconds to complete. @webjunkie 


### Low priority / side quests

- Convert the Retention, Stickiness, Path and Funnel insights.

**Comment by benjackwhite** [comment]
# Team Re-pay

**Support hero:** @daibhin  

## Retro
(skipped)

## Hang over items from previous sprint

- 🟢 Get the team rename done @benjackwhite
- 🟢 First draft of mobile replay specification @pauldambra 
- 🟢 Android recording PoC / spike @marandaneto
  - Got Android SDK sending "canvas" data for demo app


## OKR

* 📱 Mobile Replay 🟢
  - iOS rewrite looking good
  
### High priority
- Investigate whether canvas as a target makes sense for iOS (PoC) @marandaneto 
- Figure out Canvas player - either within RRweb or our own player @pauldambra 


### Low priority / side quests
- Network response payloads (would need our own branch

**Comment by robbie-c** [comment]
# Team Web Analytics

**Support hero:** @robbie-c 

## Hang over items from previous sprint
n/a

## OKR
🟢 Build the dashboards to match 80% of functionality - work underway and making good progress
⚪ Start working towards the architecture to scale to a general release - not started

### High priority
* https://github.com/PostHog/posthog/issues/18177 - should be this week
* Start opt-in feature preview & be responsive to user feedback
* Searchable pathnames, better pathname widget

### Low priority
* Change posthog-js default storage https://github.com/PostHog/posthog-js/pull/875

**Comment by EDsCODE** [comment]
# Team Data Warehouse

### OKR Q4

### Objective

**PostHog users can effortlessly bring their business data to PostHog and analyze it** 🟢

* **Key Results:**
    * Be able to import data into data warehouse in one click from most popular SaaS tools (X, Y, Z connectors built)
    * 5 happy users/teams who have imported external data
    * Data Warehouse queries are compatible with insights hogQL

**PostHog is moved over to whatever CH infrastructure we will be using for the next 1 year (at least)** 🟡 

* **Key Results:**
    * We are either on new infrastructure for EU that we host or on ClickHouse Cloud
    * Queries for large customers are performing faster
    * We can scale more easily (without James bespoke effort)
 

### Retro

**Product**
- [x] *Get the pricing and stripe connector through the door and deployed for beta users. All of it should be implemented by end of week
- [ ] Work through a few more connectors (TBD which ones are important)

**Infra**
- [ ] Complete required performance improvements with ClickHouse.cloud
- [ ] Remove EBS volumes on Prod US cluster for cost savings

### High priority

**Product**
- [ ] get billing deployed
- [ ] have a logging and monitoring dashboard setup
- [ ] stream selector interface

**Infra**

**Comment by raquelmsmith** [comment]
# Team Growth

Support hero: Raquel, then Bianca

## Retro
- [x] Fix inability to delete orgs due to foreign key on old billing tables
- [ ] Fix revenue forecasting for quarterly billing
  - Still in progress
- [ ] Keep iterating on onboarding exp until it's good
- [ ] Launch data warehouse pricing https://github.com/PostHog/billing/issues/378
  - Still in progress
- [x] Unsubscribe survey

## OKR

- ⚪  **Objective 1:** Better understand what leads to revenue
  - Reasoning: We're still a bit fuzzy about what metrics are leading indicators for revenue.
  - KR: A company scorecard with actionable insights that help us understand where to invest
  - KR: An understanding (doc or dashboard) of how people use multiple products
- ⚪ **Objective 2:** Simplify and improve the upgrade process
  - Reasoning: It's currently somewhat confusing and requires too many clicks.
  - KR: Increase the percentage of orgs who upgrade to a paid account
  - KR: Decrease or maintain the number of surprise bills that CS has to refund
  - KR: Automatically subscribe people to the products they use
- ⚪ **Bonus Objective 3:** Continue experimenting with onboarding and activation metrics
  - KR: Product-specific onboarding performs better than the old onboarding flow
  - KR: 2 experiments for Product Analytics & Session Replay activation metrics each

## This sprint
### High priority
- [x] Make sure unsub survey response is only sent if there is a response @xrdt https://github.com/PostHog/posthog/pull/18431
- [ ] Keep iterating on onboarding exp until it's good @raquelmsmith 
    - [ ] add the project switcher to the /products screen
    - [ ] make mobile friendly
    - [ ] make sure people who have existing projects aren't suddenly redirected
      -  use `snipped_onboarding_completed`
- [ ] Launch data warehouse pricing https://github.com/PostHog/billing/issues/378
  - [ ] Automatically set $500 limit on subscribe to data_warehouse @xrdt
- [ ] Make sure people aren't receiving billing limit emails every day @xrdt
- [x] Deal with the incident @raquelmsmith 
- [x] Deal with usage reports not landing in the db? -- I was using the wrong date 🤦‍♀ 


### Low priority / side quests

- [ ] Change MRR per product reporting to be calculated based on usage, not invoices
- [ ] Store an email for every billing customer, not just the subscribed ones
- [ ] Website pricing page update https://github.com/PostHog/posthog.com/issues/7056 @xrdt https://github.com/PostHog/billing/pull/398

**Comment by neilkakkar** [comment]
# Team Feature Success

**Support hero:** @neilkakkar   

Days off:
Juraj: 1 day
Neil: 1 day
Li: 2 day

## Retro

- Feature flag Environments: https://github.com/PostHog/posthog/issues/13160 - @jurajmajerik - should be ready to roll out to users in this sprint! 🥳
- New feature flag targeting options: Relative dates: https://github.com/PostHog/posthog/issues/17392, by distinct ID: https://github.com/PostHog/posthog/issues/13808 - @neilkakkar 
    - somewhat sidelined by higher prio bugs in flag matching, but this is working in app and 2 SDKs, others to follow (and then UI) - will rollover.

- Fix toolbar for flags: https://github.com/PostHog/posthog/issues/17616 https://github.com/PostHog/posthog/issues/18079 https://github.com/PostHog/posthog/issues/15481 - @liyiy -> will be done by Fri!


## Hang over items from previous sprint


---

## OKRs

1. Get 5 reference customers for surveys
2. Get reference customers for feature flags and experiments

### High priority


- Finish all 5 server SDKs for targeting - Ruby/Go @liyiy ; node/php @neilkakkar 
    - Do geoip updates when touching other server side SDKs: https://github.com/PostHog/posthog/issues/14351
    - (like https://github.com/PostHog/posthog-python/pull/98) 
- Survey user interviews & quick fixes - @liyiy  
    - Surveys: allow customising 'next' and 'close' buttons 
- Quota limiting for flags - observability - @neilkakkar 
- Scheduled rollouts of flags (and potentially everything else): https://github.com/PostHog/posthog/issues/13842 - @jurajmajerik 


### Low priority / side quests

- Generate cohort of all current users for whom flag is enabled: easy access to all exposed users (@neilkakkar )
- Stale feature flags: https://github.com/PostHog/posthog/issues/16497 (@jurajmajerik )

**Comment by davemurphysf** [comment]
# Team Pipeline


**Support hero:** Brett / Ted

## Retro



- Capture-rs is deployed and capturing traffic for team 2 from PostHog-js @xvello 
- Finish migrations of Bigquery and Postgres, and finish redshift app @tomasfarias
- Run PoE experiment in prod and figure out next steps depending on that @bretthoerner - made progress, ran it on prod, which nearly killed PG. trying to improve perf of existing queries. trying to do less back and forth.

## Hang over items from previous sprint



- - Make replicator use compose webhook and elements_chain @tiina303 / @davemurphysf 

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red
- keep the lights on
  - green, RDK gave us some issues with handling but in a better place now. 
- Batch exports migrations complete - @tomasfarias
  - green, about to migrate people over on bigquery and postgres
- Webhooks v2 @tiina303
  - New delivery system
    - yellow, nothing shipped but Dave said he had some exp with webhook systems - he could potentially ship this
  - New interface for webhooks and onevent apps
    - green, started working on UI for this

### High priority

- Capture-rs monitoriing @xvello
- Build pipeline UI @tiina303 
- Batch exports - migrate bigquery/postgres/redshift + cancel endpoint @tomasfarias @tkaemming 
- Ship PoE @bretthoerner

### Low priority / side quests

- Work with Monitoring team to write mini RFC with shape of payload and failures modes of capture @xvello 
- PostHog-js to send elements_chain as a strong @davemurphysf

**Comment by benjackwhite** [comment]
# Project Noteforce3000

## Hang over items from previous sprint

- [ ] 🔵 Dark (and tan mode) mode polish  ⭐️ @daibhin
  - Still more to do here but lots of improvements already
- [ ] 🔵 New Nav/Top Bar (child issue: https://github.com/PostHog/posthog/issues/18090) @Twixes ⭐️
  - should be there by the end of the sprint
- [x] 🔵 Cohorts -> Persons, Annotations -> Data Management (and adding any missing apps) @benjackwhite  
- [ ] 🔵 Universal Search/Command Palette @thmsobrmlr ⭐️⭐️⭐️
  - Probably rolling over into next sprint...
- [x] 🔵 Right Side Panel Notebooks @benjackwhite
- [x] 🔵 Right Side Panel Feedback @benjackwhite
- [] 🔵 Right Side Panel Docs @benjackwhite
  - In and feature flagged (waiting on website changes)
- [x] Revert project homepage to "old" mode @daibhin 
- [x] Insight fixes for Notebooks @thmsobrmlr 
  - It works!
  - .... but its not saving correctly...

### High priority

- [ ] 🔵 Dark mode final colors  ⭐️ @daibhin @corywatilo 
- [ ] 🔵 Universal Search/Command Palette @thmsobrmlr ⭐️⭐️⭐️
  - [ ] ... and get this in Notebooks (stretch goal)
- [ ] Tidying @daibhin 
  - [ ] Button styles
  - [ ] Condensed table design tweaks
  - [ ] Icons tidying 
  - [ ] Ensure the flow from start to finish works in 3000
- [ ] 🔵 Right Side Panel Activity @benjackwhite
- [ ] Fix insight saving of date ranges in notebook @thmsobrmlr 

### Side quests
- [ ] 🟣 Query Builder ⭐️⭐️⭐️ (PoC)  @corywatilo @Twixes @benjackwhite 
- [ ] 🟣 Person Feed final work (PoC-ing Canvas concept) @Twixes @daibhin 
- [ ] Insights layout for Notebooks (e.g. resizeable) @benjackwhite
